assets/prototype/domain/shared/dateRangeInput/dateDisplay.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Icon, Intent, Tag } from '@blueprintjs/core/lib/esm';
+import classNames from 'classnames';
+import { format, isValid } from 'date-fns';
+
+const FORMAT = 'EEEE, PPP';
+const FORMAT_TIME = 'EEEE, PPP p';
+
+interface DateFCProps {
+	withTime?: boolean;
+	format?: string;
+}
+
+interface DateRangeDisplayProps extends DateFCProps {
+	className?: string;
+	range: [Date, Date];
+}
+
+interface DateTagProps extends DateFCProps {
+	date: Date;
+}
+
+export const DateTag: React.FC<DateTagProps> = ({
+	date,
+	withTime = false,
+	format: formatString = withTime ? FORMAT_TIME : FORMAT,
+}): JSX.Element => {
+	if (isValid(date)) {
+		return <Tag intent={Intent.SUCCESS}>{format(date, formatString)}</Tag>;
+	} else {
+		return <Tag minimal={true}>no date</Tag>;
+	}
+};
+
+export const DateRangeDisplay: React.FC<DateRangeDisplayProps> = ({
+	className,
+	range: [start, end],
+	withTime = false,
+	format = withTime ? FORMAT_TIME : FORMAT,
```

Comment:
Not sure about inlining ternary when destructuring.
Maybe:
```suggestion
export const DateRangeDisplay: React.FC<DateRangeDisplayProps> = ({
	className,
	range: [start, end],
	withTime = false,
	...props
	format = withTime ? FORMAT_TIME : FORMAT,
}): JSX.Element => {
   const format = props.format || withTime ? FORMAT_TIME : FORMAT;
   
   return (
	<div className={classNames('docs-date-range', className)}>
		<DateTag withTime={withTime} date={start} format={format} />
		<Icon icon='arrow-right' />
		<DateTag withTime={withTime} date={end} format={format} />
	</div>
)};
```

Reply:
That was already there, I didn't add it. I agree that it will be better to move it out of destructuring. There is one more point to notice though, `withTime` is being destructured and used in the same statement, which is a bit confusing.